VMEncryption/main/HttpUtil.py

```diff
@@ -54,11 +54,11 @@ def __init__(self, logger):
     snapshot also called this. so we should not write the file/read the file in this method.
     """
 
-    def Call(self, method, http_uri, data, headers, use_https=True):
+    def Call(self, method, http_uri, data, headers, use_https=True, noProxy=False):
         try:
             uri_obj = urlparse(http_uri)
             #parse the uri str here
-            if self.proxyHost is None or self.proxyPort is None:
+            if self.proxyHost is None or self.proxyPort is None or noProxy==True:
```

Comment:
This comment is for line 76 (independent of your changes). Since it is not use_https, it should be **HTTPConnection**, not HTTPSConnection